api/fbs/v1/query.fbs

```diff
@@ -0,0 +1,121 @@
+// Licensed to Apache Software Foundation (ASF) under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Apache Software Foundation (ASF) licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+namespace v1;
+
+enum BinaryOp: byte {
+    EQ,
+    NE,
+    LT,
+    GT,
+    LE,
+    GE,
+    HAVING,
+    NOT_HAVING,
+    RANGE
+}
+
+table ByteValue {
+    value: [ubyte];
+}
+
+table Tag {
+    key: [ubyte];
+    values: [ByteValue];
```

Comment:
We should support [ByteArray] instead of [ByteValue] here. In the last online meeting, we discussed some tags that support multiple value items under a single key.

Reply:
`ByteValue` is a array of byte (i.e. an item). Then `[ByteValue]` is a multi-value item.

Since flatbuffers does not natively support vector in vector. This is a workaround.

Reply:
Let's focus on https://github.com/apache/skywalking-banyandb/pull/2#issuecomment-846840742 first. Keep our design consistent. I feel we are facing big mismatch.

And byte as value(array or not) is very dangerous, bytes type is not good at term or accurate matching, but we need that. Because byte means no encoding mechanism, we have no idea how to increase the query performance.

Reply:
> Let's focus on [#2 (comment)](https://github.com/apache/skywalking-banyandb/pull/2#issuecomment-846840742) first. Keep our design consistent. I feel we are facing big mismatch.
> 
> And byte as value(array or not) is very dangerous, bytes type is not good at term or accurate matching, but we need that. Because byte means no encoding mechanism, we have no idea how to increase the query performance.

You mean `key` or `values`?

As I've discussed with @hanahmily, the searchable tags will be given directly to the Index module while the Index module understands the type of the index (either text or number). The search pattern can be specified by `BinaryOp`.

Reply:
@wu-sheng As we discussed, the byte array is the proper data type for fields and tags. 

@lujiajing1126 sorry, didn't notice `[]` around `ByteValue`